mailet/standard/src/test/java/org/apache/james/transport/matchers/SentByMailetTest.java

```diff
@@ -0,0 +1,70 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.matchers;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collection;
+
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailAddress;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailContext;
+import org.apache.mailet.base.test.FakeMatcherConfig;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SentByMailetTest {
+
+    private SentByMailet testee;
+
+    @Before
+    public void setUp() throws Exception {
+        testee = new SentByMailet();
+        testee.init(FakeMatcherConfig.builder().matcherName("matcherName")
+            .mailetContext(FakeMailContext.defaultContext())
+            .build());
+    }
+
+    @Test
+    public void matchShouldReturnRecipientsWhenAuthUserAttributeIsPresent() throws Exception{
+        MailAddress recipient = new MailAddress("cuong.trnn@gmail.com");
+        FakeMail fakeMail = FakeMail.builder()
+            .recipient(recipient)
+            .attribute(Mail.SENT_BY_MAILET, "true")
+            .build();
+
+        Collection<MailAddress> results =  testee.match(fakeMail);
+
+        assertThat(results).containsOnly(recipient);
+    }
+
+    @Test
+    public void matchShouldNotReturnRecipientsWhenAuthUserAttributeIsAbsent() throws Exception{
+        FakeMail fakeMail = FakeMail.builder()
+            .recipients(new MailAddress("cuong.trnn@gmail.com"))
+            .build();
+
+        Collection<MailAddress> results =  testee.match(fakeMail);
+
+        assertThat(results).isNull();
```

Comment:
Because empty is better than null?

I believe it is better to return empty than null. Returning null is a bad practice.

Then we also change what is returned in SentByMailet. And if you care about being consistent with SMTPAuthSuccesfulTest then we can also return empty there.

(I made the remark because it's a newly implemented mailet)